### ISSUE TITLE
Deriving from type(Protocol) metaclass

### ISSUE THREAD

**Issue by igoras1993** [microsoft/pyright#9217]
# Description
Hi, recently encountered small issue. Deriving from `type(Protocol)` to implement a metaclass compatible with protocol metaclass is interpreted as an error by `pyright`. I might be missing something, but it seems like a small bug. Am I missing something?

## Code sample

```python
from typing import Protocol
from typing import _ProtocolMeta

class MyMeta(type(Protocol)):  # Fails
    pass

class TheSameMeta(_ProtocolMeta):  # Passes
    pass

assert _ProtocolMeta is type(Protocol)  # True
```

Running pyright against this code yields the following:

```
.../poc.py
  .../poc.py:4:14 - error: Class cannot derive from itself (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```
Obviously I don't want to import `_ProtocolMeta`, I prefer using `type(Protocol)` as `_ProtocolMeta` is not public member of typing module.

## Version
```
$ pyright --version
pyright 1.1.384
```
```
$ python3 --version
Python 3.12.7
```

**Comment by erictraut** [comment]
I agree this is a bug. Thanks for reporting it.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.387.
